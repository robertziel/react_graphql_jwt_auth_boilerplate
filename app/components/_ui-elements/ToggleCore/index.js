import React from 'react';
import PropTypes from 'prop-types';

import { InputBase, MenuItem, Select } from '@material-ui/core';

function Toggle(props) {
  let options = [];

  if (props.values) {
    options = props.values.map((value) => (
      <MenuItem key={value} value={value}>
        {(props.messages && props.messages[value]) || value}
      </MenuItem>
    ));
  }

  return (
    <Select
      className={props.className}
      value={props.value || ''}
      onChange={props.onToggle}
      input={<InputBase />}
    >
      {options}
    </Select>
  );
}

Toggle.propTypes = {
  className: PropTypes.string,
  onToggle: PropTypes.func,
  values: PropTypes.array,
  value: PropTypes.string,
  messages: PropTypes.object,
};

export default Toggle;
