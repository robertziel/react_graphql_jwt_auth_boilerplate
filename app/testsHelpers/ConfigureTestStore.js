import { memoryHistory } from 'react-router-dom';
import configureStore from 'configureStore';
import StoreAccessor from 'containers/BackendApiConnector/StoreAccessor';

export default function ConfigureTestStore() {
  const confuguredStore = configureStore({}, memoryHistory);

  StoreAccessor.store = confuguredStore.store;

  this.store = confuguredStore.store;
  this.presistor = confuguredStore.presistor;
}
