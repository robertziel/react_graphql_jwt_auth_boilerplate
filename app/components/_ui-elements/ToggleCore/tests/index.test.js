import React from 'react';
import { mount } from 'enzyme';
import { IntlProvider, defineMessages } from 'react-intl';
import { Provider } from 'react-redux';
import ConfigureTestStore from 'testsHelpers/ConfigureTestStore';

import ToggleCore from '../index';

const defaultEnMessage = 'someContent';
const defaultPlMessage = 'someOtherContent';

const messages = defineMessages({
  en: {
    id: 'en',
    defaultMessage: defaultEnMessage,
  },
  pl: {
    id: 'en',
    defaultMessage: defaultPlMessage,
  },
});

describe('<ToggleCore />', () => {
  let store;

  describe('render', () => {
    let wrapper;

    function mountWrapper(values) {
      return mount(
        <Provider store={store}>
          <IntlProvider locale="en">
            <ToggleCore values={values} messages={messages} />
          </IntlProvider>
        </Provider>,
      );
    }

    beforeAll(() => {
      store = new ConfigureTestStore().store;
    });

    it('should contain default text', () => {
      wrapper = mountWrapper(['en', 'pl']);
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should not have ToggleCoreOptions if props.values is not defined', () => {
      wrapper = mountWrapper(null);
      expect(wrapper.find('.MuiList')).toHaveLength(0);
    });
  });
});
