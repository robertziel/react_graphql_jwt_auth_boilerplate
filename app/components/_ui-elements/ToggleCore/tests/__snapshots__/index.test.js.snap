// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleCore /> render should contain default text 1`] = `"<div class=\\"MuiInputBase-root\\"><div class=\\"MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input\\" tabindex=\\"0\\" role=\\"button\\" aria-haspopup=\\"listbox\\"><span>â€‹</span></div><input aria-hidden=\\"true\\" tabindex=\\"-1\\" class=\\"MuiSelect-nativeInput\\" value=\\"\\"><svg class=\\"MuiSvgIcon-root MuiSelect-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></div>"`;
